src/shared/sam/activation.ts

```diff
@@ -97,11 +97,15 @@ async function registerServerlessCommands(ctx: ExtContext): Promise<void> {
         }),
         vscode.commands.registerCommand('aws.addSamDebugConfiguration', addSamDebugConfiguration),
         vscode.commands.registerCommand('aws.pickAddSamDebugConfiguration', codelensUtils.pickAddSamDebugConfiguration),
-        vscode.commands.registerCommand('aws.deploySamApplication', async regionNode => {
+        vscode.commands.registerCommand('aws.deploySamApplication', async arg => {
+            // `arg` is one of :
```

Comment:
Consider adding this documentation to the constructor as well.
Interested to know why you wouldn't unbox `arg` here, (maybe) simplifying the constructor?

Reply:
That splits the logic. The logic doesn't really belong in `SamDeployWizard`, but I wanted to avoid a big refactoring.